sdklab/fuzzing/fuzz_send_message.py

```diff
@@ -0,0 +1,118 @@
+drop_individual_outgoing_packets = 2
+drop_incoming_packets_until_reconnect = 3
+drop_individual_incoming_packets = 4
+raise_send_exception = 5
+raise_receive_exception = 6
+
+fuzz_type_help = """
+drop_outgoing_packets_until_reconnect = 1
+drop_individual_outgoing_packets = 2
+drop_incoming_packets_until_reconnect = 3
+drop_individual_incoming_packets = 4
+raise_send_exception = 5
+raise_receive_exception = 6
+"""
+
+
+async def main(fuzz_type):
+    device_client = IoTHubDeviceClient.create_from_connection_string(
+        test_env.DEVICE_CONNECTION_STRING, keep_alive=15
+    )
+    service_helper = ServiceHelper(
+        iothub_connection_string=test_env.IOTHUB_CONNECTION_STRING,
+        eventhub_connection_string=test_env.EVENTHUB_CONNECTION_STRING,
+        eventhub_consumer_group=test_env.EVENTHUB_CONSUMER_GROUP,
+    )
+
+    paho_fuzz_hook.add_paho_logging_hook(device_client)
+
+    try:
+
+        # Connect the device client.
+        print("connecting")
+        await device_client.connect()
+
+        # Start fuzzing after the client is connected
+        if fuzz_type == drop_outgoing_packets_until_reconnect:
+            paho_fuzz_hook.add_hook_drop_outgoing_until_reconnect(device_client, 0.05)
+        elif fuzz_type == drop_individual_outgoing_packets:
+            paho_fuzz_hook.add_hook_drop_individual_outgoing(device_client, 0.05)
+        elif fuzz_type == drop_incoming_packets_until_reconnect:
+            # probability for incoming is calculated on every byte, so it needs to be much lower
+            paho_fuzz_hook.add_hook_drop_incoming_until_reconnect(device_client, 0.001)
+        elif fuzz_type == drop_individual_incoming_packets:
+            paho_fuzz_hook.add_hook_drop_individual_incoming(device_client, 0.05)
+        elif fuzz_type == raise_send_exception:
+            paho_fuzz_hook.add_hook_raise_send_exception(device_client, 0.05)
+        elif fuzz_type == raise_receive_exception:
+            paho_fuzz_hook.add_hook_raise_receive_exception(device_client, 0.05)
+        else:
+            assert False
+
+        # TOOD: can we add device_id and module_id attributes on the client?
+        service_helper.set_identity(
+            device_id=device_client._mqtt_pipeline.pipeline_configuration.device_id,
+            module_id=device_client._mqtt_pipeline.pipeline_configuration.module_id,
+        )
+
+        print("sleeping to let eventhub consumers spin up correctly")
+        await asyncio.sleep(5)
+
+        print("sending")
+        (send_tasks, message_ids) = await queue_send_messages(device_client)
+        await asyncio.gather(*send_tasks)
+        print("done sending")
+
+        print("waiting")
+        await wait_for_messages(service_helper, message_ids)
+
+    finally:
+        print("Shutting down service helper")
+        await service_helper.shutdown()
+
+        print("Shutting down device client")
+        await device_client.shutdown()
+
+
+if __name__ == "__main__":
```

Comment:
If we're accepting user arguments via command line, I'd like to see that documented